modules/component-store/src/component-store.ts

```diff
@@ -40,6 +40,12 @@ import {
 import { isOnStateInitDefined, isOnStoreInitDefined } from './lifecycle_hooks';
 import { toSignal } from '@angular/core/rxjs-interop';
 
+// This is a global variable provided by Angular in development mode
+// (when running `serve`, for example) or by Terser (through its global definitions)
+// when running build with optimizer enabled. This allows us to tree-shake the code
+// from production bundles, ensuring that it is only called during development when needed.
+declare const ngDevMode: boolean;
```

Comment:
Thanks for this PR @arturovt! We have already considered using `ngDevMode`. However, we're not using it because it's an internal API that can be changed in any patch/minor version.

Reply:
Well, that's true that they may find reasons to remove it. However, let's consider the case where the variable is removed or renamed. The code would still maintain its backward compatibility due to the use of `isDevMode`. Since we're accessing it safely (with the `typeof` operator), the behavior wouldn't change. If the Angular team decides to remove the variable from their code, it would be a simple matter of removing it from our codebase as well.

Reply:
Have you tested this PR?

Tree shaking is not working because of the `|| isDevMode()` condition.

You can test it by executing: `yarn nx build --project=standalone-app`

Reply:
Yeah, I tested changes in another project, just build the `component-store` and linked it to `@ngrx/component-store` in another folder.

Reply:
Following what @arturovt wrote on angular/issues#51175 I'd be inclinded to think that this is the best solution right now. 

* `ngDevMode` is used - everywhere - in the angular code base, it's not going to change anytime soon. It isn't a standard constant as it's set by the CLI and only declared as a global.

Reply:
I agree, but the other side of this is that if we use this, and the bundle size changes significantly, and then it gets removed, we have no way of allowing it to be tree-shaken again unless something else is introduced. Is this bit of code that much to need this change?

No one else AFAIK, even library authors outside of the framework is using it.